### ISSUE TITLE
Making some animations to play in the background

### ISSUE THREAD

**Issue by CelticMinstrel** [CelticMinstrel/wesnoth#45]
**[Original submission](https://gna.org/bugs/index.php?5720)** by ln-zookeeper on 2006-04-07

While a recruited unit is fading in or when a unit is still moving, it's not possible to select another unit and issue orders to it, which makes playing much less fluid in non-accelerated mode, especially if there are lots of units to move. Such animations should not block UI actions, like selecting another unit, and should therefore be moved to play "in the background".

If an action (like moving, selecting, recruiting or capturing) will trigger an event, it can be known before the actual animation is started, and so if an event will trigger, then the animation can "block" UI actions like selecting another unit and moving it (like it does now whether an event triggers or not). This kind of behaviour is needed to prevent a situation where, for example, the player moves a unit into a location which triggers a name#moveto event, while at the same time selecting another unit which triggers a nameselect event.

**Comment by CelticMinstrel** [comment]
**Modified on 2008-01-20**

**martinxyz wrote:**

> From looking at the code, it already is possible to select a different unit during a move (or attack). This could trigger an event. Currently players never do this because they learned that their unit will get deselected after a move. I'm about to change this.

**Comment by CelticMinstrel** [comment]
**Modified on 2008-01-20**

**boucman wrote:**

> I don't see how this help...
> 
> 
> 
> the code is still stuck in the animation loop during the move...
> 
> 
> 
> unfortunately it will be very tricky to have mvt anims be real background anims. This is post 1.4 stuff anyhow

**Comment by CelticMinstrel** [comment]
**Modified on 2008-01-21**

**martinxyz wrote:**

> During the animation loop mouse events are still being processed the normal way by calling events::pump(). However most actions are blocked by the command_disabler. This is not &quot;true&quot; backgrounding, of course.

**Comment by CelticMinstrel** [comment]
**Modified on 2008-02-02**

**boucman changed status:** None -> Postponed

**boucman wrote:**

> complete background anims is planned for 1.5

**Comment by CelticMinstrel** [comment]
**Modified on 2010-08-28**

**eleazar wrote:**

> 1.9 has come and i still find it annoying to have to wait for the recruit animation to finish before i'm allowed to recruit/recall another unit.

**Comment by CelticMinstrel** [comment]
**Modified on 2010-08-29**

**boucman wrote:**

> it's pretty tricky from a UI pont of view...
> 
> 
> 
> the problem is that multiple animations could take place simultaneously and we need to keep WML events in the proper order, some of the happens before animations, some happen after, and they all need to look good.
> 
> 
> 
> this is esp tricky with mvt animations

**Comment by CelticMinstrel** [comment]
**Modified on 2010-08-31**

**eleazar wrote:**

> There's a strong argument that could be made that movement animations shouldn't be allowed to happen at the same time.  Otherwise it's unclear which hexes are occupied and which are not, which unit moved the shroud, etc.
> 
> 
> 
> The FR is just for &quot;some&quot; animations to play in the background.

**Comment by CelticMinstrel** [comment]
**Modified on 2012-01-29**

**shadowm changed status:** Postponed -> None
